stdlib/javascript/compile.mc

```diff
@@ -121,56 +210,104 @@ lang MExprJSCompile = MExprAst + JSProgAst
+      else never in
+    JSEObject { fields = map compileField fieldSeq }
 
-  | TmSeq {tms = tms, ty = ty, info = info} & t ->
-    -- Special handling of strings
+  | TmSeq {tms = tms} ->
     -- Check if sequence of characters, then concatenate them into a string
     if _isCharSeq tms then
       match (_charSeq2String tms) with Some str then JSEString { s = str }
-      else infoErrorExit (infoTm t) "Non-literal strings currently unsupported."
+      else never
     else
-      -- infoErrorExit (infoTm t) "Non-literal strings currently unsupported."
-      -- Else compile each expression in sequence and return a list
-      let tms: [JSExpr] = map compileExpr tms in
-      JSESeq { exprs = tms, info = info }
-
-  | TmRecordUpdate _ -> error "Record updates cannot be handled in compileExpr."
-  | TmConApp _ -> error "Constructor application in compileExpr."
-  | TmLet _ -> error "Let expressions cannot be handled in compileExpr."
-  | TmRecLets _ -> error "Recursive let expressions cannot be handled in compileExpr."
-  | TmType _ -> error "Type expressions cannot be handled in compileExpr."
-  | TmConDef _ -> error "Constructor definitions cannot be handled in compileExpr."
-  | TmMatch _ -> error "Match expressions cannot be handled in compileExpr."
-  | TmUtest _ -> error "Unit test expressions cannot be handled in compileExpr."
-  | TmExt _ -> error "External expressions cannot be handled in compileExpr."
+      JSEArray { exprs = map (compileMExpr opts) tms }
 
   -- Literals
   | TmConst { val = val } ->
     match val      with CInt   { val = val } then JSEInt   { i = val }
     else match val with CFloat { val = val } then JSEFloat { f = val }
     else match val with CChar  { val = val } then JSEChar  { c = val }
     else match val with CBool  { val = val } then JSEBool  { b = val }
+    else match compileCOp opts [] val with jsexpr then jsexpr -- SeqOpAst Consts are handled by the compile operator semantics
     else error "Unsupported literal"
+  | TmRecordUpdate _ -> error "Record updates cannot be handled in compileMExpr."
+
+
+  ----------------
+  -- STATEMENTS --
+  ----------------
+
+  | TmLet { ident = id, body = expr, inexpr = e } ->
+    -- Check if identifier is the ignore identifier (_, or [])
+    -- If so, ignore the expression unless it is a function application
+    let data = (match nameGetStr id with [] then
+      match expr with TmApp { } then -- Inline the function call
+        ([(compileMExpr opts) expr], (compileMExpr opts) e)
+      else -- Ignore the expression
+        ([], (compileMExpr opts) e)
+    else -- Normal let binding
+      ([JSEDef { id = id, expr = (compileMExpr opts) expr }], (compileMExpr opts) e)
+    ) in
+    match data with (exprs, ret) then
+      flattenBlock (JSEBlock {
+        exprs = exprs,
+        ret = ret
+      })
+    else never
+
+  | TmRecLets { bindings = bindings, inexpr = e } ->
+    -- Todo: Compile each binding as a let expression
```

Comment:
We have a convention for comments like this: https://github.com/miking-lang/miking/wiki/Conventions.
It should be formatted as `TODO(<user>, <date>): ...`